### PR TITLE
[Frontend] Warn if user `max_model_len` is greater than derived `max_model_len`

### PR DESCRIPTION
**Pull request by fialhocoelho**
#### Summary
This update addresses the handling of user-specified `MAX_SEQUENCE_LENGTH` values that exceed the model's maximum sequence length. The behavior is controlled by the environment variable `VLLM_ALLOW_LONG_MAX_MODEL_LEN`, which defaults to `0`.

@njhill @Yard1 @robertgshaw2-neuralmagic @maxdebayser I mistakenly closed the other PR while syncing with the main. Here is the link to the previous one: #5911  (to maintain comment traceability).

#### Motivation
Previously, an error was triggered when user `max_model_len` exceeded the derived value, potentially leading to unintended behavior or CUDA errors. By changing this to a warning, users are alerted without halting execution, allowing flexibility depending on their needs.

#### Notes
- Adjusted from f-string interpolation to % formatting for logging, in line with best practices (referencing [W1203](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html)).
- The change ensures compatibility with environments where logging-format-interpolation may be disabled, maintaining robustness across different configurations.
- **Default Behavior (`VLLM_ALLOW_LONG_MAX_MODEL_LEN=0`)**:
  - If the user-specified `MAX_SEQUENCE_LENGTH` is greater than the model's maximum sequence length, an error will be raised. This prevents potential incorrect model outputs or CUDA errors that could arise from exceeding the model's capabilities.
- **Optional Behavior (`VLLM_ALLOW_LONG_MAX_MODEL_LEN=1`)**:
  - If `VLLM_ALLOW_LONG_MAX_MODEL_LEN` is set to `1`, the system will allow the user-specified `MAX_SEQUENCE_LENGTH` to exceed the model's maximum. Instead of raising an error, a warning message will be logged, advising users to ensure the value is correct and within the model's context size. This provides flexibility for advanced users who understand the risks and need to override the default limit.

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -1541,15 +1542,20 @@ def _get_and_verify_max_len(
...
-                "to incorrect model outputs or CUDA errors. Make sure the "
-                "value is correct and within the model context size.")
+                "to incorrect model outputs or CUDA errors.")
+            if envs.VLLM_ALLOW_LONG_MAX_MODEL_LEN:
+                logger.warning(
+                    "%s Make sure the value is correct and within the "
+                    "model context size.", msg)
+            else:
+                raise ValueError(f"{msg} To enable this, set the env var "
+                                 "VLLM_ALLOW_LONG_MAX_MODEL_LEN=1")
```
Comment (njhill) [inline vllm/config.py:1558]: Suggested rewording
```suggestion
                raise ValueError(f"{msg} To allow overriding this maximum, set the env var "
                                 "VLLM_ALLOW_LONG_MAX_MODEL_LEN=1")
```

File: vllm/envs.py [hunk 2]
```diff
@@ -313,6 +314,15 @@ def get_default_config_root():
     # If set, vllm will skip the deprecation warnings.
     "VLLM_NO_DEPRECATION_WARNING":
     lambda: bool(int(os.getenv("VLLM_NO_DEPRECATION_WARNING", "0"))),
+
+    # If the env var VLLM_ALLOW_LONG_MAX_MODEL_LEN is set, it allows
+    # the user to specify a max sequence length greater than
+    # the model's max length.
```
Comment (njhill) [inline vllm/envs.py:320]: Suggested rewording
```suggestion
    # the max length derived from the model's config.json.
```

**Review by njhill — COMMENTED** [review]
Thanks @fialhocoelho! Just have some minor rewording suggestions.

**Review by njhill — APPROVED** [review]
Thanks @fialhocoelho

**Comment by njhill** [comment]
@fialhocoelho could you merge in the main branch again, it should help with the CI test failures.

**Comment by fialhocoelho** [comment]
Sure!

**Pull request merged**